app/vep/vcf_results.py

```diff
@@ -0,0 +1,235 @@
+""" Module for loading a VCF and parsing it into a VepResultsResponse
+object as defined in APISpecification"""
+
+from typing import Dict, Any, IO, List
+
+import vcfpy
+
+from ..vep import model
+
+TARGET_COLUMNS = [
+    "Allele",
+    "AF",
+    "Consequence",
+    "Feature",
+    "Feature_type",
+    "BIOTYPE",
+    "CANONICAL",
+    "SYMBOL",
+    "Gene",
+    "STRAND",
+    "IMPACT",
+]
+
+
+# Taken from https://github.com/Ensembl/ensembl-hypsipyle
+# main/common/file_model/variant.py#L142
+# Needs to be moved into a shared module
+def _set_allele_type(alt_one_bp: bool, ref_one_bp: bool, ref_alt_equal_bp: bool):
+    """Create a allele type for a variant based on Variation
+    teams logic using ref and largest alt allele sizes"""
+    match [alt_one_bp, ref_one_bp, ref_alt_equal_bp]:
+        case [True, True, True]:
+            allele_type = "SNV"
+            so_term = "SO:0001483"
+
+        case [True, False, False]:
+            allele_type = "deletion"
+            so_term = "SO:0000159"
+
+        case [False, True, False]:
+            allele_type = "insertion"
+            so_term = "SO:0000667"
+
+        case [False, False, False]:
+            allele_type = "indel"
+            so_term = "SO:1000032"
+
+        case [False, False, True]:
+            allele_type = "substitution"
+            so_term = "SO:1000002"
+    return allele_type, so_term
+
+
+def _get_prediction_index_map(
```

Comment:
Does this remains fixed or changes based on the output vcc ?

Reply:
This will change depending on the options provided by the user. Additional columns are added to output the results of different plugins / options